### ISSUE TITLE
Build from the source error 

### ISSUE THREAD

**Issue by xiaomingfuckeasylife** [aergoio/aergo#76]
So I installed cmake,glide , and go

```
> cmake --version
cmake version 3.14.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

```
> glide --version
glide version 0.13.2
```

```
> go version
go version go1.11.2 darwin/amd64
```

after I did all the fetching code ,  I got this error when type the make command:

```
> make
make: Circular build <- build dependency dropped.
make[1]: *** No targets specified and no makefile found.  Stop.
make: *** [Makefile] Error 2
```

**Comment by graup** [comment]
Hi, thanks for opening this issue.

Can you try to delete the "build" directory and then run `make` again? I just tried it with a clean checkout of the source and it works. I do get the same error as you when I cancel the build while it is running. We'll look into that.

**Comment by xiaomingfuckeasylife** [comment]
hi , cool , i delete the build directory , and tried again , this time i got another error :
```
[INFO]	--> Exporting gopkg.in/yaml.v2
[INFO]	Replacing existing vendor dependencies
[100%] Built target vendor
Scanning dependencies of target aergosvr
# github.com/aergoio/aergo/contract
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple LLVM version 10.0.0 (clang-1000.10.44.2)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to http://developer.apple.com/bugreporter/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg:
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /var/folders/gn/zsl4tz8n4y39h5wcsprxh1d40000gn/T/sqlite3-binding-044474.c
clang: note: diagnostic msg: /var/folders/gn/zsl4tz8n4y39h5wcsprxh1d40000gn/T/sqlite3-binding-044474.sh
clang: note: diagnostic msg: Crash backtrace is located in
clang: note: diagnostic msg: /Users/clark/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
clang: note: diagnostic msg:

********************
make[3]: *** [CMakeFiles/aergosvr] Error 2
make[2]: *** [CMakeFiles/aergosvr.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

**Comment by wrpark79** [comment]
Please check again as follows.
1. After deleting the build directory, delete the following files if exist.

CMakeCache.txt
CMakeFiles
cmake_install.cmake

2. If the Makefile has changed, revert it.
3. Run make.

If the same error occurs, do the following to upload "make.log" file.

rm -rf build
make VERBOSE = 1> ~ / make.log

**Comment by xiaomingfuckeasylife** [comment]
ha, weird, I just type make again and this time it works and the build directory is not deleted :
```
make: Circular build <- build dependency dropped.
[ 11%] Performing build step for 'libgmp'
[ 22%] Performing install step for 'libgmp'
Making install in mpn
Making install in mpz
Making install in cxx
 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/src/gmp-6.1.2/install-sh -c -d '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libgmp.la '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib'
libtool: install: /usr/bin/install -c .libs/libgmp.10.dylib /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib/libgmp.10.dylib
libtool: install: (cd /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib && { ln -s -f libgmp.10.dylib libgmp.dylib || { rm -f libgmp.dylib && ln -s libgmp.10.dylib libgmp.dylib; }; })
libtool: install: /usr/bin/install -c .libs/libgmp.lai /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib/libgmp.la
 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/src/gmp-6.1.2/install-sh -c -d '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/include'
 /usr/bin/install -c -m 644 gmp.h '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/include'

+-------------------------------------------------------------+
| CAUTION:                                                    |
|                                                             |
| If you have not already run "make check", then we strongly  |
| recommend you do so.                                        |
|                                                             |
| GMP has been carefully tested by its authors, but compilers |
| are all too often released with serious bugs.  GMP tends to |
| explore interesting corners in compilers and has hit bugs   |
| on quite a few occasions.                                   |
|                                                             |
+-------------------------------------------------------------+

[ 33%] Completed 'libgmp'
[ 88%] Built target libgmp
[ 88%] Built target liblmdb
==== Building LuaJIT 2.1.0-beta3 ====
==== Successfully built LuaJIT 2.1.0-beta3 ====
==== Installing LuaJIT 2.1.0-beta3 to /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool ====
==== Successfully installed LuaJIT 2.1.0-beta3 to /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool ====

Note: the development releases deliberately do NOT install a symlink for luajit
You can do this now by running this command (with sudo):

  ln -sf luajit-2.1.0-beta3 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/bin/luajit

[ 88%] Built target libluajit
[ 88%] Built target libtool
[100%] Built target vendor
[100%] Built target aergosvr
Scanning dependencies of target aergocli
[100%] Built target aergocli
Scanning dependencies of target aergoluac
[100%] Built target aergoluac
Scanning dependencies of target brick
[100%] Built target brick
Scanning dependencies of target build
[100%] Built target build
[ 11%] Performing build step for 'libgmp'
[ 22%] Performing install step for 'libgmp'
Making install in mpn
Making install in mpz
Making install in cxx
 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/src/gmp-6.1.2/install-sh -c -d '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libgmp.la '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib'
libtool: install: /usr/bin/install -c .libs/libgmp.10.dylib /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib/libgmp.10.dylib
libtool: install: (cd /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib && { ln -s -f libgmp.10.dylib libgmp.dylib || { rm -f libgmp.dylib && ln -s libgmp.10.dylib libgmp.dylib; }; })
libtool: install: /usr/bin/install -c .libs/libgmp.lai /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/lib/libgmp.la
 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/src/gmp-6.1.2/install-sh -c -d '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/include'
 /usr/bin/install -c -m 644 gmp.h '/Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/include'

+-------------------------------------------------------------+
| CAUTION:                                                    |
|                                                             |
| If you have not already run "make check", then we strongly  |
| recommend you do so.                                        |
|                                                             |
| GMP has been carefully tested by its authors, but compilers |
| are all too often released with serious bugs.  GMP tends to |
| explore interesting corners in compilers and has hit bugs   |
| on quite a few occasions.                                   |
|                                                             |
+-------------------------------------------------------------+

[ 33%] Completed 'libgmp'
[ 88%] Built target libgmp
[ 88%] Built target liblmdb
==== Building LuaJIT 2.1.0-beta3 ====
==== Successfully built LuaJIT 2.1.0-beta3 ====
==== Installing LuaJIT 2.1.0-beta3 to /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool ====
==== Successfully installed LuaJIT 2.1.0-beta3 to /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool ====

Note: the development releases deliberately do NOT install a symlink for luajit
You can do this now by running this command (with sudo):

  ln -sf luajit-2.1.0-beta3 /Users/clark/workspace/golang/src/github.com/aergoio/aergo/libtool/bin/luajit

[ 88%] Built target libluajit
[ 88%] Built target libtool
[100%] Built target vendor
[100%] Built target aergosvr
[100%] Built target aergocli
[100%] Built target aergoluac
[100%] Built target brick
[100%] Built target build
```

**Comment by wrpark79** [comment]
There is an ambiguous target in the Makefile for ease of development. We will fix it soon.

**Comment by wrpark79** [comment]
We did fix it at afda7b07d88f9627a56ea5d950f7bbfff0594aaa.
If you still get any error, please open issue again.
Thanks.
